### PR TITLE
Fix deleting duplicated rpaths

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Review by HexDecimal — CHANGES_REQUESTED** [review]
There are performance implications to calling `replace_signature` multiple times. It is already called too many times for the same file in the current code.

Let `delete_rpath` take multiple paths. ~~[Counter](https://docs.python.org/3/library/collections.html#collections.Counter) can be useful to track when a parameter must be given multiple times.~~ Actually just iterating over the paths one at a time is fine as long as `replace_signature` is called only once.

If a new function must be added then make it private: `def _delete_rpath`

Mind editing the docs for `get_rpaths` to clarify that duplicate rpaths may be returned?

**Comment by isuruf** [comment]
Sorry, didn't see your edit. Shall I revert the Counter changes?

**Review by HexDecimal — CHANGES_REQUESTED** [review]
I still think `_delete_rpath` should take an iterable of paths to remove (`paths: Iterable[str]`). `replace_signature` and logging would be moved into `_delete_rpath`.

Then `_remove_absolute_rpaths` would be simplified to this:
```py
_delete_rpath(
    filename,
    (rpath for rpath in get_rpaths(filename) if not _is_rpath_sanitary(rpath)),
)
```

File: delocate/tools.py [hunk 1]
```diff
@@ -858,14 +889,16 @@ def _remove_absolute_rpaths(filename: str, ad_hoc_sign: bool = True) -> None:
...
+        (
+            rpath
+            for rpath in get_rpaths(filename)
+            if not _is_rpath_sanitary(rpath)
+        ),
+        ad_hoc_sign,
+    )
+
     if ad_hoc_sign:
         replace_signature(filename, "-")
```
Comment (HexDecimal) [inline delocate/tools.py:903]: Redundant since `_delete_rpaths` handles signing.
```suggestion
```

**Review by HexDecimal — CHANGES_REQUESTED** [review]
Close to finishing. This should also be mentioned as a fix in the changelog.

File: Changelog.md [hunk 2]
```diff
@@ -23,6 +23,11 @@ rules on making a good Changelog.
 - `delocate-wheel` is now more strict with platform tags and will no longer allow
   a wheel to be incompatible with its own tags. #198
 
+### Fixed
+
+- Fixes deleting duplicated rpaths when sanitize-rpaths option is turned on.
+  [#208](https://github.com/matthew-brett/delocate/pull/208)
```
Comment (HexDecimal) [inline Changelog.md:29]: Less vague changelog item.
```suggestion
- `--sanitize-rpaths` was not removing duplicate rpaths.
  [#208](https://github.com/matthew-brett/delocate/pull/208)
```

**Review by HexDecimal — CHANGES_REQUESTED** [review]
Latest commit mentioned but did not touch the redundant `replace_signature` call.

At this point I can handle the rest if you're finished.

File: Changelog.md (hunk 2 above)
Comment (isuruf) [reply Changelog.md]: Changed this to '`--sanitize-rpaths` was failing with duplicate rpaths.' as it was failing with an exception from subprocess.

**Review by HexDecimal — APPROVED** [review]

**Comment by isuruf** [comment]
Thanks @HexDecimal for the reviews.

**Comment by HexDecimal** [comment]
I'm slightly hesitant with the recent changes and the changes I've been considering, but nothing is actually preventing a release it in its current state. I'll do a release now. I probably should've done one earlier.

**Pull request merged**